app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,21 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  skip_before_action :verify_authenticity_token, only: :github
+
+  def github
+    # You need to implement the method below in your model (e.g. app/models/user.rb)
+    @user = User.from_omniauth(request.env["omniauth.auth"])
+
+    if @user.persisted?
+      sign_in_and_redirect @user, event: :authentication # this will throw if @user is not activated
+      set_flash_message(:notice, :success, kind: "Github") if is_navigational_format?
+    else
+      flash[:error]='There was a problem signing you in through Github. Please try signing in later.'
+            redirect_to new_user_session_url
```

Comment:
Perhaps we should redirect to *root_path* here? If it fails there is probably no point in just hitting the form again? It'll presumably keep failing since the API is probably down or something.

Reply:
This can be either really, My thinking though is if someone sees a login screen again they know immediately their login failed. If we redirect to root path they may miss the error flash and think they have logged in, they will eventually realize they aren't.

Reply:
Ah OK, I didn't think of that.
Let's leave it as-is then.